src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,788 @@
+        Returns:
+            float: scan time in seconds
+        """
+        return functions.scanTime(dwell_time, points_fast, points_slow)
+
+    async def start_scan(self, simulate=False) -> None:
+        """
+        Starts a scan with the EBIC scan controlbox.
+        Send settings to the server/client to initiate scan.
+        """
+        self._dwell_time = self.get_dwell_time(self.delay)
+        self._scan_time = self.get_scan_time(self._dwell_time, self.repetition.value[0], self.repetition.value[1])
+
+        result = await self.StartScanOPCUA.call_method(
+            ua.NodeId("ScanObjectOPCUA", self.idx),  # FIXME this might not be the right format
+            ua.QualifiedName("StartScanOPCUA", self.idx),
+            self.repetition.value[0],
+            self.repetition.value[1],
+            self.oversampling.value,
+            self.numberOfChannels.value,
+            self.spp.value,
+            self.delay,
+            TRIGGER=True,
+            Simulate=simulate)
+
+    def stop_scan(self):
+        # this function is to be used in the special SPARC acq
+        pass
+
+    async def retrieve_acquired_data(self):
+        """
+        Reads a variable from the server.
+        """
+        data = await self._client.nodes.root.get_child(["0:Objects", f"{self.idx}:{DATA_VAR_NAME}"])
+        value = await data.read_value()
+        return numpy.array(list(value), dtype=numpy.float64)
+
+    async def read_variable(self, obj_name: str, var_name: str) -> Any:
+        """
+        Reads a variable from the OPC server.
+        :param obj_name:
+        :param var_name:
+        :return:
+        """
+        var = await self._client.nodes.root.get_child(f"0:Objects/{self.idx}:{obj_name}/{self.idx}:{var_name}")
+        value = await var.read_value()
+        return value
+
+    def on_repetition_change(self, value):
+        return value
+
+    def on_spp_change(self, value):
+        return value
+
+    def on_chan_num_change(self, value):
+        return value
+
+    def on_oversampling_change(self, value):
+        self.set_scan_parameters()
+
+
+class MightyEbicSimulator:
+    """
+    OPC Server class: This class is responsible for setting up the server and the state machine.
+    it can interface directly with hardware using the functions module.
+    """
+    def __init__(self):
+        """Initialize the OPC Server class."""
+        self.server = Server()
+        self.idx = None
+        self.state_machine = None
+        self.state_machine_node = None
+        self.states = {}
+        self.transitions = {}
+
+    async def setup(self) -> None:
+        """Call server async setup functions."""
+        await self.server.init()
+        self.server.set_endpoint("opc.tcp://localhost:4840/freeopcua/server/")
+
+        # set up our own namespace, not really necessary but should as spec
+        self.idx = await self.server.register_namespace("http://examples.freeopcua.github.io")
```

Comment:
You should use `NAMESPACE`, or `SIM_NAMESPACE`. And probably you want to use a namespace which is more going to match the actual device (I hope they don't keep using that example namespace!).

Reply:
I have chosen a different approach.